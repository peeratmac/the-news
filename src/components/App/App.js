import React, { Component } from 'react';
import Menu from '../Menu/Menu';
import NewsContainer from '../NewsContainer/NewsContainer';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      selectedNews: [],
      local: [],
      technology: [],
      entertainment: [],
      science: [],
      health: [],
      searchNews: []
    };
  }

  async componentDidMount() {
    const response = await fetch(
      'https://whats-new-api.herokuapp.com/api/v1/news'
    );
    const data = await response.json();
    this.setState({
      selectedNews: data.technology,
      local: data.local,
      technology: data.technology,
      entertainment: data.entertainment,
      science: data.science,
      health: data.health,
      searchedNews: data.technology
    });
  }

  changeNewsCategory = category => {
    this.setState({ selectedNews: category, searchedNews: category });
  };

  render() {
    return (
      <div className='app'>
        <Menu
          changeNewsCategory={this.changeNewsCategory}
          newsState={this.state}
        />
        <NewsContainer articles={this.state.selectedNews} />
      </div>
    );
  }
}

export default App;
