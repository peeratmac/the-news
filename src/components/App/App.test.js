import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('should match the App Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should be able to update the state of selectedNews and searchedNews', () => {
    const news = [
      {
        description:
          'Classic-rock band The Who has postponed its Denver concert, set for Sunday, Sept. 29, after an illness robbed singer Roger Daltrey of his voice, the band said on Twitter.',
        headline: 'The Who postpones Denver Concert at the Pepsi Center',
        id: 1,
        img:
          'https://theknow.denverpost.com/wp-content/uploads/2016/03/CDXXTHEWHODSC_4082x.jpg',
        url:
          'https://theknow.denverpost.com/2019/09/27/the-who-postpones-denver-concert/225182/'
      }
    ];
    const mockState = {
      selectedNews: [],
      local: [],
      technology: [],
      entertainment: [],
      science: [],
      health: [],
      searchNews: []
    };

    expect(wrapper.state()).toEqual(mockState);
    wrapper.setState({ selectedNews: news, searchedNews: news });
    expect(wrapper.state('selectedNews')).toEqual(news);
    expect(wrapper.state('searchedNews')).toEqual(news);
  });
});
