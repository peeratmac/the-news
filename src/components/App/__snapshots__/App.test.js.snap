// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the App Snapshot 1`] = `
<div
  className="app"
>
  <Menu
    changeNewsCategory={[Function]}
    newsState={
      Object {
        "entertainment": Array [],
        "health": Array [],
        "local": Array [],
        "science": Array [],
        "searchNews": Array [],
        "selectedNews": Array [],
        "technology": Array [],
      }
    }
  />
  <main>
    <SearchForm
      searchNews={[Function]}
    />
    <div
      className="news-container"
    >
      <NewsContainer
        articles={Array []}
      />
    </div>
  </main>
</div>
`;
